src/test/java/api/support/fakes/FakeOkapi.java

```diff
@@ -515,4 +522,39 @@ private String findFakeCalendarById(String servicePointId, MultiMap queries) {
       servicePointId, queries));
     return getCalendarById(servicePointId, queries).toString();
   }
+
+  private CompletableFuture<JsonObject> setEffectiveLocationIdForItem(JsonObject item) {
```

Comment:
This approach seems to derive the item effective location from the item and the holdings record. 

I thought in the mod-inventory work, you, @mreno-EBSCO and I agreed that this approach is more complicated than necessary and we only need enough to demonstrate that the value comes from the storage module.

The decision appears different for this change, please can you help me understand why.

Reply:
Yes, unfortunately there are plenty of tests that requires the `effectiveLocationId` to be set.  

The easiest way to make it work is to implement the whole location calculation logic.

Reply:
Ok, I think we should raise an issue for changing the tests to be less dependent upon the details of effective location calculation.

How much work do you think it would be to simplify this?

Reply:
Yes, there are about 400 tests that require location to be not null. I don't see how we can achieve this, maybe always set `permanentLocationId` for items, but I'm not sure if this is a true way, because in a fact it will make Item.permanentLocationId required field.

Reply:
I think setting the item permanent location on all or most of the examples is ok, in order to make the fake specify an effective location.

(The kind of kludge is an example of the coupling introduced by having business logic in the storage modules, at the moment, I don't know a way around it)

Reply:
Will raise a separate jira, it is tricky enough  to change it, because of using statics for items and non-statics for locations.